src/expr/src/vector_op/timestampz.rs

```diff
@@ -25,3 +36,125 @@ pub fn f64_sec_to_timestampz(elem: OrderedF64) -> Result<i64> {
         .to_i64()
         .ok_or(ExprError::NumericOutOfRange)
 }
+
+#[inline(always)]
+pub fn timestamp_at_time_zone(input: NaiveDateTimeWrapper, time_zone: &str) -> Result<i64> {
+    let time_zone = parse_time_zone(time_zone)?;
```

Comment:
Seems we need to parse the timezone every row, not sure whether it can be heavy. 🤔

Some possible optimizations:
- If the second expr is a `Constant`, use another optimized implementation like #5130.
- Use a thread-local memoization / LRU cache.

Reply:
The parsing is a lookup of a static and perfect hash table (crate `phf`), where hash is done in a case-insensitive manner. So memoization / cache has no advantage here.

It is still true that most cases this is a constant, though.

Reply:
Cool. Then there should be no problem.

Reply:
Updated the function name to `lookup_time_zone`.